### PR TITLE
Document OoB in z_message_nes

### PR DESCRIPTION
**Pull request by garrettjoecox**

### CODE REVIEW

File: src/code/z_message_nes.c
Comment (engineer124) [inline src/code/z_message_nes.c:1578]: In most cases, this issue was artificial and could just be fixed by fixing the stack, which we've done a few times for these decode functions now. However, `s16 digits[4]; // spC8` needs to be at stack 0xC8, and `s16 numLines; // spD0` needs to be at stack 0xD0, so there's not really any wiggle room for stack. Plus, replacing `numLines` with `digits[5]` doesn't work since `numLines` carries over in the loop.

So I think this might be an authentic OoB? I'd like a bit more time with this function and maybe other experienced decompers can take a look too for a second opinion. But interesting find. Thanks

Comment (engineer124) [reply src/code/z_message_nes.c:1578]: Yeah, I'm not seeing an easy way to fix this. This comment is fine then.

Something to take in consideration that we are trying to get everybody who has contributed / wants to contribute to MM decomp to agree to licensing the repo before merging PRs from people who has not yet agreed to it. It is better explained here: https://github.com/louist103/mm-licensing/issues/1.

Could you please read about it and post in that issue if you agree or not. Thanks!

**Comment by ginomeelectra202** [comment]
!ts1 elpmet eht ni mih dnif ot deen i tuB

**Comment by yemenalcyone230** [comment]
?pets a gnissim I ma ro gub a siht sI .stnevE gniognO ni llits si tseuq eht tub tsrif gab bmob reggib eht thguob dna feiht eht deppots I yenom evas oT

**Comment by altoidmerope294** [comment]
Ok so this is becoming a problem, I'm at the part where I'm supposed to sneak around above the garden on the platforms, but every single time I either get hit by one of the mad scrubs, or end up killing it, or fall.I was actually crying earlier because I was so p*ssed off by it.SOMEONE HALPHow do I get past this without getting hit by one of the mad scrubs and having to do it all over again...

**Comment by hirottau302** [comment]
?od I dluohs tahw .ti evomer ot woh wonk t'nod I tub ti no nus a htiw kcolb gib siht s'ereht tub ydelem ssenipme fo ygele eht dnif ot evah I erew trap eht ta ma I

**Comment by yemenalcyone230** [comment]
?emit ni kcab og ot evah syawla I lliW ?ereht evas ot ti si dab woh dna od eutats lwo eht seod tahW ?evas ot woh wonk ot tnaw I

**Comment by hirottau302** [comment]
All I know is is that it has a holographic cartridge cover.

File: src/code/z_message_nes.c
Comment (garrettjoecox) [reply src/code/z_message_nes.c:1578]: @engineer124 Sorry for the delay, agreed that license 👍

**Review by engineer124 — APPROVED** [review]

**Review by AngheloAlf — APPROVED** [review]

**Review by hensldm — APPROVED** [review]

**Pull request merged**
